src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,76 @@
+package duke;
+
+
+import duke.commands.Command;
+import duke.commands.DeadlineCommand;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.EventCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.TodoCommand;
+
+import java.time.LocalDateTime;
+
+/**
+ * Parser class to parse commands given by the user.
+ */
+public class Parser {
+    private static boolean isCommandMatch(String line, String match) {
+        return line.startsWith(match) && (line.length() == match.length() || line.charAt(line.indexOf(match) + match.length()) == ' ');
+    }
+
+    /**
+     * Converts the user's input into a command object.
+     * The command object would contain the code to carry out the relevant actions intended by the user.
+     * @param line user's input
+     * @return relevant command object
+     * @throws DukeException
+     */
+    public static Command parse(String line) throws DukeException {
```

Comment:
A pretty long stretch of code, would it be possible to abstract further?

Reply:
Will look into abstracting further in future updates. Thank you.